indexing-service/src/test/java/org/apache/druid/indexing/common/IndexTaskClientTest.java

```diff
@@ -0,0 +1,153 @@
+        id -> TaskLocation.create(id, -2, -2)
+    )) {
+      expectedException.expect(MalformedURLException.class);
+      expectedException.expectMessage("Invalid port number :-2");
+
+      indexTaskClient.submitRequestWithEmptyContent(
+          "taskId",
+          HttpMethod.GET,
+          "test",
+          null,
+          true
+      );
+    }
+  }
+
+  @Test
+  public void retryOnChannelException() throws IOException
+  {
+    final HttpClient httpClient = EasyMock.createNiceMock(HttpClient.class);
+    EasyMock.expect(httpClient.go(EasyMock.anyObject(), EasyMock.anyObject(), EasyMock.anyObject()))
+            .andReturn(Futures.immediateFailedFuture(new ChannelException("IndexTaskClientTest")))
+            .times(2);
+    EasyMock.expect(httpClient.go(EasyMock.anyObject(), EasyMock.anyObject(), EasyMock.anyObject()))
+            .andReturn(
+                Futures.immediateFuture(
+                    new FullResponseHolder(
+                        HttpResponseStatus.OK,
+                        new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK),
+                        new StringBuilder()
+                    )
+                )
+            )
+            .once();
+    EasyMock.replay(httpClient);
+    try (IndexTaskClient indexTaskClient = buildIndexTaskClient(httpClient, id -> TaskLocation.create(id, 8000, -1))) {
+      final FullResponseHolder response = indexTaskClient.submitRequestWithEmptyContent(
+          "taskId",
+          HttpMethod.GET,
+          "test",
+          null,
+          true
+      );
+      Assert.assertEquals(HttpResponseStatus.OK, response.getStatus());
+    }
+  } 
+
+  private IndexTaskClient buildIndexTaskClient(HttpClient httpClient, Function<String, TaskLocation> taskLocationProvider)
+  {
+    final TaskInfoProvider taskInfoProvider = new TaskInfoProvider()
+    {
+      @Override
+      public TaskLocation getTaskLocation(String id)
+      {
+        return taskLocationProvider.apply(id);
+      }
+
+      @Override
+      public Optional<TaskStatus> getTaskStatus(String id)
+      {
+        return Optional.of(TaskStatus.running(id));
+      }
+    };
+    return new TestIndexTaskClient(
+        httpClient,
+        objectMapper,
+        taskInfoProvider,
+        new Duration(1000),
+        "indexTaskClientTest",
+        1,
+        numRetries
+    );
+  }
+
+  private static class TestIndexTaskClient extends IndexTaskClient
+  {
+    private TestIndexTaskClient(
+        HttpClient httpClient,
+        ObjectMapper objectMapper,
+        TaskInfoProvider taskInfoProvider,
+        Duration httpTimeout,
+        String callerId,
+        int numThreads,
+        long numRetries
+    )
+    {
+      super(httpClient, objectMapper, taskInfoProvider, httpTimeout, callerId, numThreads, numRetries);
+    }
+
+    @Override
+    protected void checkConnection(String host, int port)
+    {
+      // do nothing
+    }
+  }
+}
```

Comment:
needs newline (this is why Travis is failing)

Reply:
Thanks, fixed!